extensions/pyRevitTools.extension/checks/CADAUDIT_check.py

```diff
@@ -0,0 +1,200 @@
+        return cadinstances, len(cadinstances)
+    else:
+        print("No CAD instances found in the {}.".format("active view" if active_view_only else "model"))
+        script.exit()
+
+# Manage Flexform cancel using .NET System.Windows RoutedEventArgs Class
+class ButtonClass(Window): # to handle button event (https://stackoverflow.com/questions/54756424/using-the-on-click-option-for-a-revit-rpw-flexform-button)
+    @staticmethod
+    def cancel_clicked(sender, e):
+        window = Window.GetWindow(sender)
+        window.close()
+        script.exit()
+
+# Add radio button functionality to RPW Flexforms
+class RadioButton(RpwControlMixin, Controls.RadioButton):
+    """
+    Windows.Controls.RadioButton Wrapper
+
+    >>> RadioButton('Label')
+    """
+    def __init__(self, name, radio_text, default=False, **kwargs):
+        """
+        Args:
+            name (``str``): Name of control. Will be used to return value
+            radio_text (``str``): RadioButton label Text
+            default (``bool``): Sets IsChecked state [Default: False]
+            wpf_params (kwargs): Additional WPF attributes
+        """
+        self.Name = name
+        self.Content = radio_text
+        self.IsChecked = default
+        self.set_attrs(top_offset=0, **kwargs)
+
+    @property
+    def value(self):
+        return self.IsChecked
+    
+def get_cad_site(cad_inst):
+    """ A CAD's location site cannot be got from the Shared Site parameter 
+        cad_inst.Name returns the site name with a 'location' prefix (language-specific, eg 'emplacement' in French)"""
+    cad_site_name = cad_inst.Name
+    cad_site_name.replace("location", "-")
+    return cad_site_name
+
+
+def get_user_input():
+    """ create RPW input FlexForm for user choice of collection mode (coll_mode) whole model or just active view """
+    flexform_comp = [Label("Audit CAD instances:")]
+    flexform_comp.append(RadioButton("model", "in this project", True, GroupName="grp")) # GroupName implemented in class through kwargs
+    flexform_comp.append(RadioButton("active_view", "in only the active view", False, GroupName="grp"))
+    flexform_comp.append(Separator())
+    flexform_comp.append(Button("CANCEL", on_click=ButtonClass.cancel_clicked))
+    flexform_comp.append(Button("OK"))
+
+    user_input = FlexForm("Audit of CAD instances in the current project", flexform_comp, Width=500, Height=200) # returns a FlexForm object
+    user_input.show()
+    user_input_dict = user_input.values # This is a dictionary
+    if not bool(user_input_dict): # check if dict not empty (a bool of non-empty dict is True)
+        script.exit()
+
+    return user_input_dict
+
+def get_load_stat(cad, is_link):
+    """ Loaded status from the import instance's CADLinkType """
+    cad_type = doc.GetElement(cad.GetTypeId()) # Retreive the type from the instance
+    
+    if is_link:
+        exfs = cad_type.GetExternalFileReference()
+        status = exfs.GetLinkedFileStatus().ToString()
```

Comment:
to compare enumerations, it is better not to convert them to string `if status == LinkedFileStatus.Loaded`, [see docs](https://www.revitapidocs.com/2022/410b48f8-2a32-3d54-492c-c9a9fe4030ca.htm)

Reply:
I was getting the string value to be able to return it as a message

Reply:
agree with @dosymep, not the most elegant way/
will merge anyway for now